### PR TITLE
[ALOY-1629] Upgrade to babel 7 packages

### PR DESCRIPTION
**Pull request by ewanharris**
https://jira.appcelerator.org/browse/ALOY-1629

This updates all babel packages to the new 7.x versions. This looks to be achievable with no changes on our side, I have migrated our code-frame usage to use the new `codeFrameColumns` function however as the previous one is now [deprecated](https://babeljs.io/docs/en/next/babel-code-frame.html#upgrading-from-prior-versions).

I've run through the ~alloy unit tests~ (borked on this, didn't realise these were used in the tests files directly), ks/movies, as well as 2 other apps and have seen no issues so far.

### CODE REVIEW

File: Alloy/commands/compile/ast/controller.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 var U = require('../../../utils'),
-	babylon = require('babylon'),
-	types = require('babel-types'),
-	generate = require('babel-generator').default,
-	traverse = require('babel-traverse').default;
+	babylon = require('@babel/parser'),
+	types = require('@babel/types'),
+	generate = require('@babel/generator').default,
+	traverse = require('@babel/traverse').default;
```
Comment (brentonhouse) [inline Alloy/commands/compile/ast/controller.js]: @ewanharris - After this line `const { Hub, NodePath } = traverse;` I am getting `undefined` for `Hub`.  Is this related to the `.default`?

Comment (janvennemann) [reply Alloy/commands/compile/ast/controller.js]: Yes, `Hub` is a separate export: `const Hub = require('@babel/traverse').Hub;` or `import { Hub } from '@babel/traverse';`

Comment (brentonhouse) [reply Alloy/commands/compile/ast/controller.js]: Ah!  The error doesn't happen until merged w/ latest master.   After you merge, you can change line to something like this and it works then: 
```
const { Hub, NodePath } = require('@babel/traverse');
```

**Comment by ewanharris** [comment]
@brentonhouse, updated the PR to handle the changes to Hub in `@babel/traverse`

**Comment by brentonhouse** [comment]
@ewanharris  - Are we good to merge this now?

**Comment by ewanharris** [comment]
@brentonhouse, this should be good to go once it's rubber stamped and approved.

**Pull request merged**